### ISSUE TITLE
[FEATURE] Add no-redundant-role rule

### ISSUE THREAD

**Issue by yeonjuan** [yeonjuan/html-eslint#487]
**Is your feature request related to a problem? Please describe.**

In HTML, many elements already have an implicit ARIA role defined by the specification.
Specifying the same role explicitly using the role attribute is redundant and provides no additional accessibility benefit.

### Incorrect
```html
<button role="button">Click</button>
<nav role="navigation"></nav>
<main role="main"></main>
```

### Proposal

- Introduce a new rule: no-redundant-role
- This rule would: Report cases where an explicit role matches the element’s implicit ARIA role
